src/net/connection/quality.rs

```diff
@@ -25,17 +21,12 @@ impl NetworkQualityMeasurer {
 
     /// This will calculate the round trip time (rtt) from the given acknowledgement.
     /// Where after it updates the rtt from the given connection.
-    pub fn update_connection_rtt(
+    pub fn get_rtt(
         &self,
-        connection: &mut RwLockWriteGuard<VirtualConnection>,
-        ack_seq: u16,
-    ) {
-        let smoothed_rrt = {
-            let congestion_data = connection.congestion_avoidance_buffer.get_mut(ack_seq);
-            self.get_smoothed_rtt(congestion_data)
-        };
-
-        connection.rtt = smoothed_rrt;
+        congestion_data: Option<&mut CongestionData>
+    ) -> f32 {
+        let smoothed_rrt = self.get_smoothed_rtt(congestion_data);
```

Comment:
just have this function return `self.get_smoothed_rtt` 😄 

no need for the let binding.